src/userbase-server/app.js

```diff
@@ -526,6 +575,62 @@ exports.listUsersWithPagination = async function (req, res) {
   }
 }
 
+exports.listUsersForDatabaseWithPagination = async function (req, res) {
+  let logChildObject
+  try {
+    const databaseId = req.params.databaseId
+    const nextPageToken = req.query.nextPageToken
+
+    logChildObject = { ...res.locals.logChildObject, databaseId, nextPageToken }
+    logger.child(logChildObject).info('Listing users from Admin API')
+
+    _validateDatabaseId(databaseId)
+
+    // get database stuff
+    const db = await dbController.findDatabaseByDatabaseId(databaseId)
+    if (!db) throw {
+      status: statusCodes['Not Found'],
+      error: { message: 'Database not found.' }
+    }
+    const ownerId = db['owner-id']
+
+    // get user stuff from db owner
+    const owner = await userController.getUserByUserId(ownerId)
+    const appId = owner['app-id']
+
+    const lastEvaluatedKey = nextPageTokenToLastEvaluatedKey(
+      nextPageToken,
+      (lastEvaluatedKey) => _validateListUsersLastEvaluatedKey(lastEvaluatedKey, appId)
+    )
+
+    const admin = res.locals.admin
+    const adminId = admin['admin-id']
+
+    const app = await getAppByAppId(appId)
+    _validateAppResponseToGetApp(app, adminId, logChildObject)
+
+    const usersResponse = await _getUsersForDbQuery(db['database-id'], lastEvaluatedKey)
```

Comment:
Can be Promise.all()'d with `getAppByAppId` above

Reply:
Done